nav2_collision_monitor/include/nav2_collision_monitor/velocity_polygon.hpp

```diff
@@ -0,0 +1,134 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_COLLISION_MONITOR__VELOCITY_POLYGON_HPP_
+#define NAV2_COLLISION_MONITOR__VELOCITY_POLYGON_HPP_
+
+#include <string>
+#include <vector>
+
+#include "rclcpp/rclcpp.hpp"
+#include "geometry_msgs/msg/polygon_stamped.hpp"
+#include "tf2_ros/buffer.h"
+
+#include "nav2_util/lifecycle_node.hpp"
+#include "nav2_collision_monitor/types.hpp"
+
+namespace nav2_collision_monitor
+{
+
+/**
+ * @brief Velocity polygon class.
+ * This class contains all the points of the polygon and
+ * the expected condition of the velocity based polygon.
+ */
+class VelocityPolygon
+{
+public:
+  /**
+   * @brief VelocityPolygon constructor
+   * @param node Collision Monitor node pointer
+   * @param polygon_name Name of main polygon
+   * @param velocity_polygon_name Name of velocity polygon
+   */
+  VelocityPolygon(
+    const nav2_util::LifecycleNode::WeakPtr & node,
+    const std::string & polygon_name,
+    const std::string & velocity_polygon_name,
+    const std::shared_ptr<tf2_ros::Buffer> tf_buffer,
+    const std::string & base_frame_id,
+    const tf2::Duration & transform_tolerance);
+  /**
+   * @brief VelocityPolygon destructor
+   */
+  virtual ~VelocityPolygon();
+
+  /**
+   * @brief Supporting routine obtaining velocity polygon specific ROS-parameters
+   * @return True if all parameters were obtained or false in failure case
+   */
+  bool getParameters();
+
+  /**
+   * @brief Check if the velocities and direction is in expected range.
+   * @param cmd_vel_in Robot twist command input
+   * @return True if speed and direction is within the condition
+   */
+  bool isInRange(const Velocity & cmd_vel_in);
+
+  /**
+   * @brief Check if the velocities and direction is in expected range.
+   * @param cmd_vel_in Robot twist command input
+   * @return True if speed and direction is within the condition
+   */
+  std::vector<Point> getPolygon();
+
+protected:
+  // ----- Variables -----
+
+  /// @brief Collision Monitor node
+  nav2_util::LifecycleNode::WeakPtr node_;
+  /// @brief Collision monitor node logger stored for further usage
+  rclcpp::Logger logger_{rclcpp::get_logger("collision_monitor")};
+
+  /**
+   * @brief Dynamic polygon callback
+   * @param msg Shared pointer to the polygon message
+   */
+  void polygonCallback(geometry_msgs::msg::PolygonStamped::ConstSharedPtr msg);
+
+  /**
+   * @brief Updates polygon from geometry_msgs::msg::PolygonStamped message
+   * @param msg Message to update polygon from
+   */
+  void updatePolygon(geometry_msgs::msg::PolygonStamped::ConstSharedPtr msg);
+
+  // Basic parameters
+  /// @brief Points of the polygon
+  std::vector<Point> poly_;
+  /// @brief Name of polygon
+  std::string polygon_name_;
+  /// @brief Topic for dynamic polygon subscriber
+  std::string polygon_sub_topic_;
```

Comment:
Remove parameter, make it set to what we discussed in the comments. If someone wants to remap, they can always do that at launch time. There hits a point where its not beneficial to saturate people with parameters they'll likely never change